package/MDAnalysis/coordinates/chemfiles.py

```diff
@@ -0,0 +1,335 @@
+
+
+        .. _formats: http://chemfiles.org/chemfiles/latest/formats.html
+        """
+        self.filename = filename
+        self.n_atoms = n_atoms
+        if mode != "a" and mode != "w":
+            raise IOError("Expected 'a' or 'w' as mode in ChemfilesWriter")
+        self._file = Trajectory(self.filename, mode, chemfiles_format)
+        self._closed = False
+        if topology is not None:
+            if isinstance(topology, string_types):
+                self._file.set_topology(topology)
+            else:
+                topology = self._topology_to_chemfiles(topology, n_atoms)
+                self._file.set_topology(topology)
+
+    def close(self):
+        """Close the trajectory file and finalize the writing"""
+        if hasattr(self, "_closed") and not self._closed:
+            self._file.close()
+            self._closed = True
+
+    def write(self, obj):
+        """Write object `obj` at current trajectory frame to file.
+
+        Topology for the output is taken from the `obj` or default
+        to the value of the `topology` keyword supplied to the
+        :class:`ChemfilesWriter` constructor.
+
+        Parameters
+        ----------
+        obj : Universe or AtomGroup
+            The :class:`~MDAnalysis.core.groups.AtomGroup` or
+            :class:`~MDAnalysis.core.universe.Universe` to write.
+        """
+        if hasattr(obj, "atoms"):
+            if hasattr(obj, 'universe'):
+                # For AtomGroup and children (Residue, ResidueGroup, Segment)
+                ts_full = obj.universe.trajectory.ts
+                if ts_full.n_atoms == obj.atoms.n_atoms:
+                    ts = ts_full
+                else:
+                    # Only populate a time step with the selected atoms.
+                    ts = ts_full.copy_slice(atoms.indices)
+            elif hasattr(obj, 'trajectory'):
+                # For Universe only --- get everything
+                ts = obj.trajectory.ts
+        else:
+            if isinstance(obj, base.Timestep):
+                ts = obj
+                topology = None
+            else:
+                raise TypeError("No Timestep found in obj argument")
+
+        frame = self._timestep_to_chemfiles(ts)
+        frame.topology = self._topology_to_chemfiles(obj, len(frame.atoms))
+        self._file.write(frame)
+
+    def write_next_timestep(self, ts):
+        """Write timestep object into trajectory.
+
+        Parameters
+        ----------
+        ts: TimeStep
+        """
+        frame = self._timestep_to_chemfiles(ts)
+        self._file.write(frame)
+
+    def _timestep_to_chemfiles(self, ts):
+        """
+        Convert a Timestep to a chemfiles Frame
+        """
+        frame = Frame()
+        frame.resize(ts.n_atoms)
+        if ts.has_positions:
+            frame.positions[:] = ts.positions[:]
+        if ts.has_velocities:
+            frame.add_velocities()
+            frame.velocities[:] = ts.velocities[:]
```

Comment:
forces are not supported by chemfiles? I do not mind but it should be documented in the class doc string.

Reply:
No, they are not for now, they might be added when we figure out the best way to do so. I'll mention it in the docstring!